src/middlewared/middlewared/plugins/disk_/wipe.py

```diff
@@ -45,20 +45,29 @@ async def wipe_quick(self, dev, size=None):
 
     @accepts(
         Str('dev'),
-        Str('mode', enum=['QUICK', 'FULL', 'FULL_RANDOM']),
-        Bool('synccache', default=True),
+        Dict(
+            'options',
+            Str('mode', enum=['QUICK', 'FULL', 'FULL_RANDOM'], required=True),
+            Bool('synccache', default=True),
+            Dict('swap_removal_options', default=None, additional_attrs=True),
+        ),
     )
     @job(lock=lambda args: args[0])
-    async def wipe(self, job, dev, mode, sync):
+    async def wipe(self, job, dev, options):
```

Comment:
Its a public method, I would like to avoid changing API, making it backward compatible whenever possible.